src/patterns/patterns.py

```diff
@@ -599,3 +599,286 @@ def make_file_developer_df(
             sorted_hot_files = hot_files[sorted_dev_list]
         self.top_N_map = sorted_hot_files
         return sorted_hot_files, stats_df
+
+    def extract_directories(self):
+        print("INFO: Extracting head directories from filepaths...")
+        extracted_col = self.commit_data["filepath"]
+        df = pd.DataFrame(extracted_col)
+        filepaths = df["filepath"].tolist()
+        
+        # extracting the name of head directory from each filepath
```

Comment:
What does that mean exactly? Maybe include an example in the comment?

Reply:
Done.